maven-artifact/src/main/java/org/apache/maven/artifact/versioning/ComparableVersion.java

```diff
@@ -353,15 +359,15 @@ public int getType() {
 
         @Override
         public boolean isNull() {
-            return (comparableQualifier(value).compareTo(RELEASE_VERSION_INDEX) == 0);
+            return value == null || value.isEmpty();
         }
 
         /**
          * Returns a comparable value for a qualifier.
-         *
+         * <p>
          * This method takes into account the ordering of known qualifiers then unknown qualifiers with lexical
          * ordering.
-         *
+         * <p>
          * just returning an Integer with the index here is faster, but requires a lot of if/then/else to check for -1
```

Comment:
From here to the end is implementation detail. It's a good comment but it isn't needed in API doc. Move this part to a regular comment.

Reply:
Done.